pages/options.js

```diff
@@ -0,0 +1,251 @@
+      leftIcon: new p.FAIcon("fa-caret-up"),
+      tooltip: "Move Filter Up",
+    });
+    upButton.id = "upButton-"+i,
+    
+    upButton.registerCallback("move-up", "click", function(widget) {
+      saveList(filterList.moveUp(filterList[+widget.id.split("-")[1]]));
+    });
+    
+    
+    var downButton = new p.Button({
+      textVisible: false,
+      leftIcon: new p.FAIcon("fa-caret-down"),
+      tooltip: "Move Filter Down",
+    });
+    downButton.id = "downButton-"+i,
+    
+    downButton.registerCallback("move-down", "click", function(widget) {
+      saveList(filterList.moveDown(filterList[+widget.id.split("-")[1]]));
+    });
+    
+    listItem.addChild(upButton);
+    listItem.addChild(downButton);
+    listItem.addChild(listButton);
+    
+    layout.filterListBox.addChild(listItem);
+  }
+  
+  var addButton = new p.Button({
+    text: "Add Filter",
+    tooltip: "Add a New Filter",
+    leftIcon: new p.FAIcon("fa-plus"),
+  });
+  addButton.addClass("addButton");
+  addButton.registerCallback("add-item", "click", function(widget) {
+    filterList.push({
+      src: "",
+      dest: "",
+      matchProtocol: false,
+    });
+    buildFilterList(filterList.length-1);
+  });
+  
+  var resetButton = new p.Button({
+    text: "Reset",
+    tooltip: "Reset Filter List",
+    leftIcon: new p.FAIcon("fa-exclamation-triangle"),
+  });
+  
+  resetButton.addClass("resetButton");
+  resetButton.registerCallback("reset-list", "click", function(widget) {
+    filterList = $.extend([], defaultFilterList);
+    saveList();
+  });
+  
+  
+  p.domInsert(addButton, document.getElementById("controlButtons"));
+  p.domInsert(resetButton, document.getElementById("controlButtons"));
+
+}
+
+function buildDetails(item) {
+  layout.detailBox.empty();
+  layout.applyButton ? layout.applyButton.destroy() : null;
+  layout.deleteButton ? layout.deleteButton.destroy() : null;
+  
+  if(item) {
+    var srcField = new p.TextField({
+      value: item.src,
+      tooltip: "The URL to Block or Redirect",
+    });
+    
+    var srcLabel = new p.Label({
+      text: "Source:",
+      forInput: srcField,
+      tooltip: "The URL to Block or Redirect",
+    });
+    
+    
+    var destField = new p.TextField({
+      value: item.dest,
+      tooltip: "Redirect to This URL (Leave blank to block)",
+    });
+    
+    var destLabel = new p.Label({
+      text: "Destination:",
+      forInput: destField,
+      tooltip: "Redirect to This URL (Leave blank to block)",
+    });
+    
+    
+    var protocolSwitch = new p.Switch({
+      value: item.matchProtocol,
+      tooltip: "Check the URL's Protocol (http://, https://, etc.) when Testing for This Match.",
+    });
+    
+    var protocolLabel = new p.Label({
+      text: "Match Protocol:",
+      forInput: protocolSwitch,
+      tooltip: "Check the URL's Protocol (http://, https://, etc.) when Testing for This Match.",
+    });
+    
+    layout.applyButton ? layout.applyButton.destroy() : null;
```

Comment:
Isn't this redundant? See [line #137](https://github.com/Nateowami/Backloader/pull/1/files#diff-7f00fc25e8e4066be27c7becea3631e2R137), and again on [line #201](https://github.com/Nateowami/Backloader/pull/1/files#diff-7f00fc25e8e4066be27c7becea3631e2R201). Haven't they already been removed?